cmake/dependencies.cmake

```diff
@@ -113,13 +113,30 @@ endif()
 ################################################################################
 # GPU Support
 
-if(MICM_ENABLE_CUDA)
-  find_package(CUDA REQUIRED)
-  enable_language(CUDA)
-endif()
+if(NOT ${MICM_GPU_TYPE} STREQUAL "None")
+  set(cuda_arch_map_a100 80)
+  set(cuda_arch_map_v100 70)
+  set(cuda_arch_map_h100 90a)
+  set(cuda_arch_map_h200 90a)
+  set(cuda_arch_map_b100 95)
+  set(cuda_arch_map_b200 95)
+  set(cuda_arch_map_ all-major)
+  # Setting CUDAARCHS does not override CMAKE_CUDA_ARCHITECTURES or CUDA_ARCHITECTURES
+  # until the current process has returned to the caller site in CMake.
+  set(ENV{CUDAARCHS} ${cuda_arch_map_${MICM_GPU_TYPE}})
```

Comment:
So this logic won't handle the GPU type other than NVIDIA gpus. Would the following one be more appropriate?
```
if(${MICM_GPU_TYPE} STREQUAL "v100")
  set(ENV{CUDAARCHS} 70)
elseif (${MICM_GPU_TYPE} STREQUAL "a100")
  set(ENV{CUDAARCHS} 80)
elseif (${MICM_GPU_TYPE} STREQUAL "h100" OR ${MICM_GPU_TYPE} STREQUAL "h200")
  set(ENV{CUDAARCHS} 90a)
elseif (${MICM_GPU_TYPE} STREQUAL "b100" OR ${MICM_GPU_TYPE} STREQUAL "b200")
  set(ENV{CUDAARCHS} 95)
else()
  message(FATAL_ERROR "Unsupported GPU type: ${MICM_GPU_TYPE}.")
endif()
```

Reply:
I like that the architecture value is listed once here but I think we want to avoid this for the time being because we are going to have to do this for each new gpu type that we want to support and this might get verbose.

We will also need to refactor the current implementation anyway when we move to supporting Intel and AMD and I suspect this will change significantly then as well with newer CMake support.

Reply:
Thanks for the clarification. That is clear to me now.